toolkit/tools/pkg/imagecustomizerlib/customizeutils.go

```diff
@@ -338,26 +339,62 @@ func enableOrDisableServices(services imagecustomizerapi.Services, imageChroot *
 func loadOrDisableModules(modules imagecustomizerapi.Modules, imageChroot *safechroot.Chroot) error {
 	var err error
 
+	// Handle loading modules
+	modulesToLoadMap := make(map[string]bool)
+	var modulesToLoad []string
 	for _, module := range modules.Load {
-		logger.Log.Infof("Loading kernel module (%s)", module.Name)
-		moduleFileName := module.Name + ".conf"
-		moduleFilePath := filepath.Join(imageChroot.RootDir(), "/etc/modules-load.d/", moduleFileName)
-		err = file.Write(module.Name, moduleFilePath)
-		if err != nil {
-			return fmt.Errorf("failed to write module load configuration: %w", err)
+		logger.Log.Infof("Loading kernel module %s", module)
+		modulesToLoad = append(modulesToLoad, module)
+		modulesToLoadMap[module] = true
+	}
+
+	var modulesWithOptions []string
+	for module, moduleOptions := range modules.Options {
+		if !modulesToLoadMap[module] {
+			// Load module in options list if it is not specified in load list
+			logger.Log.Infof("Loading kernel module %s", module)
```

Comment:
I find this part of the design a bit confusing. 
so specifying `Options` can substitute for `Load`? Shouldn't we merge the 2 concepts then?

Reply:
Based on offline discussion, let's not auto-load if Options are specified.